--- a/windows/ble_peripheral_plugin.h	2026-08-27 00:48:32.009947905 +0530
+++ b/windows/ble_peripheral_plugin.h	2026-08-27 00:48:50.597128681 +0530
@@ -84,6 +84,11 @@
         // BluetoothLe
         Radio bluetoothRadio{nullptr};
         BluetoothAdapter adapter{nullptr};
+        BluetoothLEAdvertisementPublisher advertisementPublisher{nullptr};
+        winrt::event_token publisherStatusChangedToken{};
+        bool publisherStarted{false};
+        void Publisher_StatusChanged(BluetoothLEAdvertisementPublisher const &sender, BluetoothLEAdvertisementPublisherStatusChangedEventArgs const &args);
+        std::string PublisherStatusToString(BluetoothLEAdvertisementPublisherStatus status);
 
         winrt::fire_and_forget InitializeAdapter();
         winrt::fire_and_forget AddServiceAsync(const BleService &service);
